bolts-tasks/src/main/java/bolts/Task.java

```diff
@@ -47,11 +47,54 @@
    */
   public static final Executor UI_THREAD_EXECUTOR = AndroidExecutors.uiThread();
 
+  /**
+   * Interface for handlers invoked when a failed <tt>Task</tt> is about to be
+   * finalized, but the exception has not been consumed.
+   *
+   * <p>The handler will execute in the GC thread, so if the handler needs to do
+   * anything time consuming or complex it is a good idea to fire off a Task
+   * to handle the exception.
+   *
+   * @see #getUnobservedExceptionHandler
+   * @see #setUnobservedExceptionHandler
+   */
+  public interface UnobservedExceptionHandler {
+    /**
+     * Method invoked when the given task has an unobserved exception.
+     * <p>Any exception thrown by this method will be ignored.
+     * @param t the task
+     * @param e the exception
+     */
+    void unobservedException(Task<?> t, UnobservedTaskException e);
```

Comment:
This currently matches `Thread.UncaughtExceptionHandler`'s pattern: https://developer.android.com/reference/java/lang/Thread.UncaughtExceptionHandler.html